EPlast/EPlast.BLL/Services/EmailSending/NewPlastMemberEmailGreeting.cs

```diff
@@ -0,0 +1,83 @@
+﻿using EPlast.BLL.Interfaces;
+using EPlast.DataAccess.Entities;
+using EPlast.DataAccess.Repositories;
+using Microsoft.AspNetCore.Identity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EPlast.BLL.Services
+{
+    public class NewPlastMemberEmailGreeting : INewPlastMemberEmailGreeting
+    {
+        private readonly IEmailSendingService _emailSendingService;
+        private readonly IRepositoryWrapper _repoWrapper;
+        private readonly UserManager<User> _userManager;
+
+        public NewPlastMemberEmailGreeting(IRepositoryWrapper repoWrapper,
+                                           UserManager<User> userManager,
+                                           IEmailSendingService emailSendingService)
+        {
+            _repoWrapper = repoWrapper;
+            _userManager = userManager;
+            _emailSendingService = emailSendingService;
+        }
+
+        public async Task<bool> NotifyNewPlastMembersAsync()
+        {
+            try
+            {
+                (await GetNewPlastMembersAsync()).ToList().ForEach(async (user) => await SendEmailGreetingForNewPlastMemberAsync(user.Email));
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+            return true;
```

Comment:
```suggestion
        public async Task NotifyNewPlastMembersAsync()
        {
                (await GetNewPlastMembersAsync()).ToList().ForEach(async (user) => await SendEmailGreetingForNewPlastMemberAsync(user.Email));

```
Could it be like this ?

Reply:
Yes, corrected